### PR TITLE
Spec update

### PR DESCRIPTION
**Pull request by corngood**
Some fixes required after updating the spec.

There's still some work needed to write extension enums properly.

### CODE REVIEW

**Comment by expipiplus1** [comment]
Looks pretty good.

What's should the policy be on empty modules, for example `src/Graphics/Vulkan/AMD/ShaderTrinaryMinmax.hs`. I'm not too keen on keeping them in as it gives the impression that they have some support.

File: generate/src/Write/Type/Define.hs [hunk 1]
```diff
@@ -28,20 +28,23 @@ import           Write.WriteMonad
...
+          Right uint32_t = C.cIdentifierFromString "uint32_t"
+      in do tellRequiredNames (S.toList requiredNames)
+            hsElemType <- cTypeToHsType (C.TypeSpecifier (C.Specifiers [] [] [])
+                                         (C.TypeName uint32_t))
+            let hsType = foldr TyFun hsElemType (hsElemType <$ boundNames)
+            pure [qc|{hsName} :: {prettyPrint hsType}
 {hsName} {hsep (fromString <$> boundNames)} = {prettyPrint expression}
 |]
+    _ -> pure mempty
+    -- symTab -> error $ "writeDefine: invalid symTab " ++ show symTab ++ " for " ++ dName d
```
Comment (expipiplus1) [inline generate/src/Write/Type/Define.hs]: Should this comment be removed?

Comment (corngood) [inline generate/src/Write/Type/Define.hs]: Yep

**Comment by corngood** [comment]
Take a look at #23.  It gets some more stuff exporting from extensions, so those modules won't actually be empty.  At the very least they should have their extension name #define.

**Comment by expipiplus1** [comment]
Superseded by the 2.0.0.0 branch. I'm really sorry this didn't end up getting merged :(

**Pull request closed without merging**